src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -188,18 +194,26 @@ public String getQuestionResultStatisticsHtml(
             String participantIdentifier = entry.getKey();
             String name = bundle.getNameForEmail(participantIdentifier);
             String teamName = bundle.getTeamNameForEmail(participantIdentifier);
+            String userAverageExcludingSelfText =
+                    getAverageExcludingSelfText(df, recipientRanksExcludingSelf, entry.getKey());
+            String fragmentTemplateToUse = isAvgExclSelfShown
+                    ? FormTemplates.RANK_RESULT_STATS_RECIPIENTFRAGMENT_SELF_RESPONSE
+                    : FormTemplates.RANK_RESULT_STATS_RECIPIENTFRAGMENT;
 
-            fragments.append(Templates.populateTemplate(FormTemplates.RANK_RESULT_STATS_RECIPIENTFRAGMENT,
+            fragments.append(Templates.populateTemplate(fragmentTemplateToUse,
                     Slots.RANK_OPTION_VALUE, SanitizationHelper.sanitizeForHtml(name),
                     Slots.TEAM, SanitizationHelper.sanitizeForHtml(teamName),
                     Slots.RANK_RECIEVED, ranksReceived,
-                    Slots.RANK_AVERAGE, df.format(average)));
+                    Slots.RANK_AVERAGE, df.format(average),
+                    Slots.RANK_EXCLUDING_SELF_AVERAGE, userAverageExcludingSelfText));
 
         }
 
-        return Templates.populateTemplate(FormTemplates.RANK_RESULT_RECIPIENT_STATS,
-                                                             Slots.RANK_OPTION_RECIPIENT_DISPLAY_NAME, "Recipient",
-                                                             Slots.FRAGMENTS, fragments.toString());
+        String templateToUse = isAvgExclSelfShown
+                ? FormTemplates.RANK_RESULT_RECIPIENT_STATS_SELF_RESPONSE
```

Comment:
Try to get rid of all these ternary operators. Ideally, set `fragmentTemplateToUse` and `templateToUse` in the same `if` statement, near where `recipientRanksExcludingSelf` is being computed.

Of course, you should keep them all somewhat visually near to where they are being used, but there is no need to define them literally the line before where they are being used.